cli/azd/pkg/input/progress_log.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package input
+
+import (
+	"bufio"
+	"log"
+	"strings"
+	"sync"
+
+	tm "github.com/buger/goterm"
+)
+
+/* progressLog implements the io.Writer interface and writes bytes applying a framed style like
```

Comment:
Just +1'ing some previous feedback from @weikanglim about how wonderfully commented this file is.  It's really helpful to see the design here!